### DISCUSSION TITLE
Question for making dll file for External Soil Model(ESM)

### DISCUSSION THREAD

**Discussion by zorroxt1118** [Anura3D/Anura3D_OpenSource#200] [Q&A]
Hello, I understand that I need to form the dll file for soil model such as strain-softening. There is a section in tutorial manual 2023 5.5 about how to form the dll file. However, for someone have few experience coding in Fortran, the section seems to be brief. I noticed that  there is a .for file **'ExternalSoilModel.for'** in the **src** folder. I am a bit confused that whether this  'ExternalSoilModel.for' file is same to the esm example in the tutorial manual. Thank you.

**Comment by stig-cz** [comment]
Hi, I have the same problem as you, do you have a solution?

**Reply by pmarveggio** [reply]
@stig-cz Try with the new software release and let us know if it works!
